demos/python_demos/common/images_capture.py

```diff
@@ -0,0 +1,114 @@
+import os
+import copy
+
+import cv2
+
+
+class InvalidInput(Exception):
+    pass
+
+
+class ImagesCapture:
+
+    def __init__(self, loop):
+        self.loop = loop
+
+    def read():
+        raise NotImplementedError
+
+
+class ImreadWrapper(ImagesCapture):
+
+    def __init__(self, input, loop):
+        super().__init__(loop)
+        self.image = cv2.imread(input, cv2.IMREAD_COLOR)
+        self.can_read = True
+        if self.image is None:
+            raise InvalidInput
+
+    def read(self):
+        if self.loop:
+            return copy.deepcopy(self.image)
+        if self.can_read:
+            self.can_read = False
+            return copy.deepcopy(self.image)
+        return None
+
+
+class DirReader(ImagesCapture):
+
+    def __init__(self, input, loop):
+        super().__init__(loop)
+        self.dir = input
+        if not os.path.isdir(self.dir):
+            raise InvalidInput
+        self.names = sorted(os.listdir(self.dir))
+        if not self.names:
+            raise InvalidInput
+        self.file_id = 0
+        read_images = 0
+        for name in self.names:
+            filename = os.path.join(self.dir, name)
+            image = cv2.imread(filename, cv2.IMREAD_COLOR)
+            if image is not None:
+                read_images += 1
```

Comment:
This counter is not needed. Return instead. The idea is to accept the input once you can make sure there is at least one image.

Reply:
done